guides/release/tutorial/autocomplete-component.md

```diff
@@ -79,25 +79,24 @@ The `handleFilterEntry` action will apply the search term filter to the list of
 
 ```javascript {data-filename=app/components/list-filter.js}
 import Component from '@ember/component';
+import { action } from '@ember/object';
 
-export default Component.extend({
-  classNames: ['list-filter'],
-  value: '',
+export default class ListFilterComponent extends Component {
+  classNames = 'list-filter';
```

Comment:
So, we should actually convert this to a GlimmerComponent. You can check out the [edition guide](https://deploy-preview-455--ember-guides.netlify.com/release/upgrading/editions/#toc_glimmer-components) to see what the differences are. In this example, we would update the template like so:


```hbs
<div class="list-filter">
  {{input
    value=this.value
    key-up=this.handleFilterEntry
    class="light"
    placeholder="Filter By City"
  }}
  {{yield this.results}}
</div>
```

This moves the `classNames` to be a part of the template, because Glimmer components don't have a wrapping element. We also remove the `{{action}}` helper from around the action, since it's no longer necessary (`@action` does the same thing). We would then update the class like this:

```js
import Component from '@glimmer/component';
import { tracked } from '@glimmer/tracking';
import { action } from '@ember/object';

export default class ListFilterComponent extends Component {
  @tracked value = '';
  @tracked results;

  constructor() {
    super(...arguments);
    this.args.filter('').then((results) => this.results = results);
  }

  @action
  handleFilterEntry() {
    let filterInputValue = this.value;
    let filterAction = this.args.filter;
    filterAction(filterInputValue).then((filterResults) => this.results = filterResults);
  }
}
```

Here we're importing from `@glimmer/component` instead of `@ember/component`, which is the new base class. We also have to use `this.args` to refer to arguments because of this, so we have to update `this.filter` to `this.args.filter` in general. Finally, Glimmer components don't have `this.set`, so we either have to use `Ember.set`, or convert to tracked props. I went for the latter here.

This PR is really looking great, thanks for being understanding, I know there are a lot of changes and they're hard to work through and not all that transparent yet. This is going to be very helpful in getting all this info out there 😄

Reply:
Oh this is great. Thanks. I understand perfectly now, I think. And I'm glad for the clarity about wrapping tags, too, since I'm sick of dealing w/ classNames and tagNames in my component.jses anyway.